common/src/main/java/com/wynntils/features/overlays/CustomPlayerListFeature.java

```diff
@@ -0,0 +1,289 @@
+                    0,
+                    ROLL_WIDTH,
+                    Texture.PLAYER_INFO_OVERLAY.height(),
+                    0,
+                    0,
+                    ROLL_WIDTH,
+                    Texture.PLAYER_INFO_OVERLAY.height(),
+                    Texture.PLAYER_INFO_OVERLAY.width(),
+                    Texture.PLAYER_INFO_OVERLAY.height());
+        }
+
+        private void renderPlayerNames(PoseStack poseStack, float categoryStart, List<StyledText> players) {
+            for (int i = 0; i < players.size(); i++) {
+                int x = i / 19;
+                int y = i % 19;
+
+                float xPos = getRenderX() + ROLL_WIDTH + 12 + (DISTANCE_BETWEEN_CATEGORIES * x);
+                float yPos = categoryStart + 14 + (10 * y);
+                FontRenderer.getInstance()
+                        .renderText(
+                                poseStack,
+                                players.get(i),
+                                xPos,
+                                yPos,
+                                CustomColor.fromChatFormatting(ChatFormatting.BLACK),
+                                HorizontalAlignment.LEFT,
+                                VerticalAlignment.MIDDLE,
+                                TextShadow.NONE);
+            }
+        }
+
+        private void renderCategoryTitle(PoseStack poseStack, String name, float currentDist, float categoryStart) {
+            FontRenderer.getInstance()
+                    .renderText(
+                            poseStack,
+                            StyledText.fromString(name),
+                            currentDist,
+                            categoryStart,
+                            CustomColor.fromChatFormatting(ChatFormatting.BLACK),
+                            HorizontalAlignment.CENTER,
+                            VerticalAlignment.MIDDLE,
+                            TextShadow.NONE);
+        }
+
+        private void renderBackground(PoseStack poseStack) {
+            RenderUtils.drawTexturedRect(
+                    poseStack,
+                    Texture.PLAYER_INFO_OVERLAY.resource(),
+                    getRenderX() + ROLL_WIDTH,
+                    getRenderY(),
+                    0,
+                    Texture.PLAYER_INFO_OVERLAY.width() - ROLL_WIDTH,
+                    Texture.PLAYER_INFO_OVERLAY.height(),
+                    ROLL_WIDTH,
+                    0,
+                    Texture.PLAYER_INFO_OVERLAY.width() - ROLL_WIDTH,
+                    Texture.PLAYER_INFO_OVERLAY.height(),
+                    Texture.PLAYER_INFO_OVERLAY.width(),
+                    Texture.PLAYER_INFO_OVERLAY.height());
+        }
+
+        @Override
+        protected void onConfigUpdate(ConfigHolder configHolder) {
+            animationPercentage.setOpeningDuration(Duration.of(openingDuration.get(), ChronoUnit.MILLIS));
+        }
+    }
+
+    private static final class AnimationPercentage {
```

Comment:
Perhaps even this could be moved `com/wynntils/utils/render/type` although it doesn't fully fit there either.